File: dener8/programmers_다음큰숫자.py
def count(num):
    remain = []
    while num:
        remain.append(num % 2)
        num //= 2
    cnt = 0
    for i in remain:
        if i == 1:
            cnt += 1
    return cnt


def solution(n):
    answer = 0
    cnt = count(n)
    for i in range(n + 1, 1000001):
        if cnt == count(i):
            return i

    return answer

File: myunghyunnero/programmers_다음큰숫자.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n) {
    int answer = 0;
    string str = "";
    int cnt = n+1;
    int ansNum =0;
    while(n!=0){
        if(n%2==1) ansNum++;
        n/=2;
    }
    while(cnt<=1000000){
        int num = cnt;
        int oneCnt = 0;
        while(num!=0){
            if(num%2==1) oneCnt++;
            num/=2;
        }
        if(ansNum == oneCnt){
            answer=cnt;
            break;
        }
        cnt++;
    }
    
    return answer;
}